File: assets/code/add_performance.rs
use std::time::Instant;

fn perf(array0: &[i32], array1: &[i32], array2: &mut [i32]) {
    for i in 0..array0.len() {
        array2[i] = array0[i] + array1[i];
    }
}

fn main() {
    println!("enter main");

    let size: usize = 1_000_000_000;

    let array0 = vec![0i32; size];
    let array1 = vec![0i32; size];
    let mut array2 = vec![0i32; size];

    println!("start");
    let t0 = Instant::now();

    perf(&array0, &array1, &mut array2);

    let duration = t0.elapsed().as_secs_f32();
    let gb_per_second = (size as f32 * 4.0) / 1e9 / duration;

    println!("{:.2} GB/s", gb_per_second);
}

File: assets/code/add_performance.py
import time

def perf(array0, array1, array2):
    for i in range(len(array0)):
        array2[i] = array0[i] + array1[i]

def main():
    print("enter main")

    size = 10_000_000
    array0 = [0] * size
    array1 = [0] * size
    array2 = [0] * size

    print("start")
    t0 = time.time()

    perf(array0, array1, array2)

    t1 = time.time()
    duration = t1 - t0
    gb_per_second = (size * 4) / 1e9 / duration

    print(f"{gb_per_second:.2f} GB/s")

if __name__ == "__main__":
    main()
